dpnp/backend/extensions/indexing/CMakeLists.txt

```diff
@@ -38,12 +38,12 @@ if(_dpnp_sycl_targets)
     target_compile_options(
         ${python_module_name}
         PRIVATE
-        -fsycl-targets=${_dpnp_sycl_targets}
+        ${_dpnp_sycl_target_compile_options}
```

Comment:
Should we do the same for each CMakeList.txt per every extension?

Reply:
@antonwolfy mkl extensions require slightly different flags than our own extensions. 
I have added new variables `_dpnp_sycl_mkl_target_compile_options` and `_dpnp_sycl_mkl_target_link_options` for them.